### ISSUE TITLE
Improve POJO documentation

### ISSUE THREAD

**Issue by maurever** [h2oai/h2o-3#16196]
Actualize this part of the doc: https://docs.h2o.ai/h2o/latest-stable/h2o-docs/productionizing.html#pojo-quick-start

- add information we already have not supported POJO (if it is true?) or actualize the list of algorithms that do not support it. 

cc: @wendycwong , @valenad1, @hannah-tillman

**Comment by wendycwong** [comment]
@maurever :  You are right that about POJO not being supported by new algorithms.  POJOs for algos that we support have not been changed or updated and we in general discourage the use of POJOs.

The code to generate POJO for models that support it can be found in toJavaPredictBody and this code has not been changed much when new features are added to the algos.

A list of algos that do not support POJOs will be a good idea to list.

For algos that support POJOs, please warn users that they only support features that exist at or before 2015????  (not sure when we decided not to continue updating POJOs.  It is probably around the same time when Michalk starts working at H2O).
